datarobot_batch_scoring/batch_scoring.py

```diff
@@ -412,23 +416,53 @@ def __iter__(self):
                                'we lost {} records'.format(
                                    batch.id, len(batch.data)))
                 continue
-            hook = partial(self._response_callback, batch=batch)
 
             if self.fast_mode:
                 chunk_formatter = fast_to_csv_chunk
             else:
                 chunk_formatter = slow_to_csv_chunk
 
-            data = chunk_formatter(batch.data, batch.fieldnames)
-            self._ui.debug('batch {} transmitting {} bytes'
-                           .format(batch.id, len(data)))
-            yield requests.Request(
-                method='POST',
-                url=self.endpoint,
-                headers=self.headers,
-                data=data,
-                auth=(self.user, self.api_token),
-                hooks={'response': hook})
+            todo = [batch]
+            while todo:
+                batch = todo.pop(0)
+                data = chunk_formatter(batch.data, batch.fieldnames)
+                if len(data) < self.max_batch_size:
+                    self._ui.debug('batch {}-{} transmitting {} bytes'
+                                   .format(batch.id, batch.rows, len(data)))
+
+                    hook = partial(self._response_callback, batch=batch)
+
+                    yield requests.Request(
+                        method='POST',
+                        url=self.endpoint,
+                        headers=self.headers,
+                        data=data,
+                        auth=(self.user, self.api_token),
+                        hooks={'response': hook})
+                else:
+                    self._ui.debug('batch {}-{} is too long: {} bytes,'
+                                   ' splitting' .format(batch.id, batch.rows,
+                                                        len(data)))
+
+                    if batch.rows < 2:
+                        self._ui.error('batch {} is single row but bigger '
+                                       'than limit, skipping, we lost {} '
+                                       'records'.format(batch.id,
+                                                        len(batch.data)))
+                        continue
+
+                    split_point = int(batch.rows/2)
```

Comment:
Should we warn or fail the script at the beginning we estimate that the --n_samples will be far more than 10MB?  Maybe that's an edge case since we really want people to use auto_sample now